src/components/ThemeEngine/StyleEngine.tsx

```diff
@@ -0,0 +1,76 @@
+import { CSS_PREFIX } from '../constants';
+import { kebabCase } from 'lodash';
+import React, { ReactNode, useMemo } from 'react';
+
+import { Root as PortalRoot } from '@radix-ui/react-portal';
+
+/**
+ * Validates a CSS color variable.
+ * @param value The value to validate
+ */
+const isColor = (value: string) => {
+  const s = new Option().style;
+  s.background = value;
+  return s.background !== '';
+};
+
+/**
+ * Adds children to document.head via Radix Portal.
+ * @param children The children to add to document.head
+ */
+const HeadPortal = ({ children }: { children: ReactNode }) => {
```

Comment:
Seems like `HeadPortal` is `React.FC`

Reply:
Yeah I probably should use `FC` - I'll make a note of it.